banzai/utils/fits_utils.py

```diff
@@ -119,38 +121,40 @@ def open_fits_file(filename):
     -----
     This is a wrapper to astropy.io.fits.open but funpacks the file first.
     """
-    base_filename, file_extension = os.path.splitext(os.path.basename(filename))
+    base_filename, file_extension = os.path.splitext(get_filename_from_info(file_info))
+    path_to_file = get_local_path_from_info(file_info, runtime_context)
+
+    need_to_download = False
+    if not os.path.isfile(path_to_file):
+        need_to_download = True
+
     if file_extension == '.fz':
         with tempfile.TemporaryDirectory() as tmpdirname:
-            output_filename = os.path.join(tmpdirname, base_filename)
-            os.system('funpack -O {0} {1}'.format(output_filename, filename))
-            hdulist = fits.open(output_filename, 'readonly')
+            output_filepath = os.path.join(tmpdirname, base_filename)
+            if need_to_download:
+                downloaded_filepath = download_from_s3(file_info, tmpdirname, runtime_context)
+                os.system('funpack -O {0} {1}'.format(output_filepath, downloaded_filepath))
```

Comment:
Shouldn't we just make the downloaded path the same as the usual path?

Reply:
`downloaded_filepath` is the `temp_dir/{filename}`, which `funpack` will operate on. This is returned by the `download_from_s3` method as the path to where the file was downloaded to. `output_filepath` is where `funpack` will write the unpacked file to.

In both cases, `funpack` is writing the output file (`.fits` - funpacked) to the `output_filepath`